### PR TITLE
enhancement(enriching): Add select parameter to enrichment VRL functions

### PR DESCRIPTION
**Pull request by StephenWakely**
Ref #8815 

This adds the select parameter to the enrichment table VRL functions. This limits the columns returned by the function which makes it easier to merge directly into an object.

For example:

```
   . |= get_enrichment_table_record!("file", {"id": lookup}, select: ["first_name", "last_name", "email"])
```

If no `select` parameter is specified all columns are returned.

I will update the docs in #8817 .

### CODE REVIEW

**Comment by StephenWakely** [comment]
There are a ton of irrelevant commits being list here - efa4382 is the only one that is really relevant. I think I started these changes from the wrong branch..

File: lib/enrichment/src/find_enrichment_table_records.rs [hunk 1]
```diff
@@ -54,6 +59,28 @@ impl Function for FindEnrichmentTableRecords {
...
+        let select = arguments
+            .optional_array("select")?
+            .map(|select| {
+                select
+                    .into_iter()
+                    .map(|expr| {
+                        let err = || vrl_core::function::Error::ExpectedStaticExpression {
+                            keyword: "select",
+                            expr: expr.clone(),
+                        };
```
Comment (jszwedko) [inline lib/enrichment/src/find_enrichment_table_records.rs:71]: Not a huge deal given I imagine this parameter would usually be static, but does this parameter need to be static? It seems like it is something that wouldn't be terribly expensive to evaluate at runtime.

**Review by jszwedko — APPROVED** [review]
Nice! Just one comment about whether the select needed to be static or not. Otherwise this looks great.

File: lib/enrichment/src/find_enrichment_table_records.rs (hunk 1 above)
Comment (StephenWakely) [reply lib/enrichment/src/find_enrichment_table_records.rs:71]: Yes, that's a good point. I think it should be dynamic. I can see people wanting to calculate the columns based on some criteria on occasion.

**Review by JeanMertz — APPROVED** [review]
I did add a few comments, but otherwise this LGTM 👍

File: lib/enrichment/src/lib.rs [hunk 2]
```diff
@@ -54,6 +55,7 @@ pub trait Table: DynClone {
         &self,
         case: Case,
         condition: &'a [Condition<'a>],
+        select: Option<&[String]>,
```
[the remark this answers is not in the document]
Comment (StephenWakely) [reply lib/enrichment/src/lib.rs:58]: We could, and initially it did, but there is little point as it means another iteration and allocation over the list taking a `Vec<String>` to a `Vec<&str>`.

File: benches/enrichment_tables_file.rs [hunk 3]
```diff
@@ -93,7 +93,7 @@ fn benchmark_enrichment_tables_file(c: &mut Criterion) {
             |(file, condition, expected)| {
                 assert_eq!(
                     Ok(expected),
-                    file.find_table_row(Case::Sensitive, condition, Some(index))
+                    file.find_table_row(Case::Sensitive, condition, None, Some(index))
```
[the remark this answers is not in the document]
Comment (JeanMertz) [reply benches/enrichment_tables_file.rs:96]: I personally wouldn't use an enum, but a regular struct to contain all the options. It'll provide for better argument naming.

**Pull request merged**